show-geodesic-path-between-two-points/src/main/java/com/esri/arcgismaps/sample/showgeodesicpathbetweentwopoints/MainActivity.kt

```diff
@@ -0,0 +1,197 @@
+import kotlinx.coroutines.launch
+import kotlin.math.roundToInt
+
+class MainActivity : AppCompatActivity() {
+
+    private val TAG = MainActivity::class.java.simpleName
+
+    // set up data binding for the activity
+    private val activityMainBinding: ActivityMainBinding by lazy {
+        DataBindingUtil.setContentView(this, R.layout.activity_main)
+    }
+
+    // set up data binding for the mapView
+    private val mapView by lazy {
+        activityMainBinding.mapView
+    }
+
+    // shows the distance information as a textview
+    private val infoTextView by lazy {
+        activityMainBinding.infoTextView
+    }
+
+    // a blue marker symbol for the location points
+    private val locationMarkerSymbol by lazy {
+        SimpleMarkerSymbol(
+            SimpleMarkerSymbolStyle.Circle,
+            Color.blue,
+            10f
+        )
+    }
+
+    // the actual marker graphic for the starting location
+    private val startingLocationMarkerGraphic by lazy {
+        Graphic().apply {
+            symbol = locationMarkerSymbol
+            geometry = startingPoint
+        }
+    }
+
+    // marker graphic for the destination, location information
+    // populated on tap
+    private val endLocationMarkerGraphic by lazy {
+        Graphic().apply {
+            symbol = locationMarkerSymbol
+        }
+    }
+
+    // the geodesic path represented as line graphic
+    private val geodesicPathGraphic by lazy {
+        val lineSymbol = SimpleLineSymbol(SimpleLineSymbolStyle.Dash, Color.blue, 5f)
+        Graphic().apply {
+            symbol = lineSymbol
+        }
+    }
+
+    // the unit of distance measurement in kilometers
+    private val unitsOfMeasurement = LinearUnit(LinearUnitId.Kilometers)
+
+
+    // starting location for the distance calculation
+    private val startingPoint = Point(-73.7781, 40.6413, SpatialReference.wgs84())
+
+
+    // creates a graphic overlay to draw all graphics
+    private val graphicsOverlay = GraphicsOverlay()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        // authentication with an API key or named user is
+        // required to access basemaps and other location services
+        ArcGISEnvironment.apiKey = ApiKey.create(BuildConfig.API_KEY)
+        lifecycle.addObserver(mapView)
+
+        // create and add a map with a navigation night basemap style
+        val map = ArcGISMap(BasemapStyle.ArcGISImageryStandard)
+        // configure mapView assignments
+        mapView.apply {
+            this.map = map
+            // add the graphics overlay to the mapview
+            graphicsOverlays.add(graphicsOverlay)
+        }
+
+        // add all our marker graphics to the graphics overlay
+        graphicsOverlay.graphics.add(startingLocationMarkerGraphic)
+        graphicsOverlay.graphics.add(endLocationMarkerGraphic)
+        graphicsOverlay.graphics.add(geodesicPathGraphic)
```

Comment:
```suggestion
        graphicsOverlay.graphics.addAll(listOf(startingLocationMarkerGraphic,
            endLocationMarkerGraphic,
            geodesicPathGraphic))
```
Use `addAll` when adding multiple graphics